### ISSUE TITLE
ERROR! conflicting action statements: name, set_fact

### ISSUE THREAD

**Issue by AuroraLi** [ansible/ansible#37302]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Ansible playbook

##### ANSIBLE VERSION

```
ansible 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/xxx/ansible/lib/python2.7/site-packages']
  ansible python module location = /usr/xxx/ansible/lib/python2.7/site-packages/ansible
  executable location = /usr/xxx/ansible/bin/ansible
  python version = 2.7.13 (default, Feb  8 2017, 06:30:30) [GCC 4.4.7 20120313 (Red Hat 4.4.7-16)]
```

##### CONFIGURATION

ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=60s
DEFAULT_KEEP_REMOTE_FILES(env: ANSIBLE_KEEP_REMOTE_FILES) = True
DEFAULT_LOG_PATH(/etc/ansible/ansible.cfg) = /var/log/ansible.log
DEFAULT_MODULE_PATH(/etc/ansible/ansible.cfg) = [u'/usr/xxx/ansible/lib/python2.7/site-packages']
DEFAULT_SCP_IF_SSH(/etc/ansible/ansible.cfg) = True
DEFAULT_SSH_TRANSFER_METHOD(/etc/ansible/ansible.cfg) = scp
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
PARAMIKO_LOOK_FOR_KEYS(/etc/ansible/ansible.cfg) = False
PARAMIKO_RECORD_HOST_KEYS(/etc/ansible/ansible.cfg) = False

##### OS / ENVIRONMENT

Red Hat Enterprise Linux Server release 6.8 (Santiago)

##### SUMMARY

Ansible playbook throws Ansible Parser Error when tasks include name and any other ansible modules, or ansible modules and register
The playbook is able to run if no name or register claimed.
This Ansible is running in a virtual environment. OS default python is 2.6, venv python version is 2.7
There might be issues with ansible setup? But I am not able to find out. 

##### STEPS TO REPRODUCE

Create a playbook file and run the file using ansible-playbook

```yaml
---
- hosts: fortinet
  #gather_facts: true
  #connection: local
  #any_errors_fatal: true
  name: FortiGate Implementation
  vars:
    ansible_python_interpreter: /usr/xxx/ansible/bin/python
  vars_files:
    - var.yml


  tasks:
    - name: set vdom
      set_fact: vdom=PSMCC
      when: Location=="ON"

```

##### EXPECTED RESULTS

TASK [set vdom] **********************************************************************************************************************************************
ok: [x.x.x.x]

##### ACTUAL RESULTS

Throws error: confilcting action statements: name, set_fact

```
ansible-playbook 2.4.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/usr/xxx/ansible/lib/python2.7/site-packages']
  ansible python module location = /usr/xxx/ansible/lib/python2.7/site-packages/ansible
  executable location = /usr/xxx/ansible/bin/ansible-playbook
  python version = 2.7.13 (default, Feb  8 2017, 06:30:30) [GCC 4.4.7 20120313 (Red Hat 4.4.7-16)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
Parsed /etc/ansible/hosts inventory source with ini plugin
ERROR! conflicting action statements: name, set_fact

The error appears to have been in '/usr/xxx/modify_loop.yml': line 14, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: set vdom
      ^ here


The error appears to have been in '/usr/xxx/modify_loop.yml': line 14, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: set vdom
      ^ here

exception type: <class 'ansible.errors.AnsibleParserError'>
exception: conflicting action statements: name, set_fact

The error appears to have been in '/usr/xxx/modify_loop.yml': line 14, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  tasks:
    - name: set vdom
      ^ here

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/playbook](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
do you have a file named 'name.*' in your library/ directory?

**Comment by alikins** [comment]
This sounds like it may be same issue that https://github.com/ansible/ansible/pull/34649 addresses.

**Comment by AuroraLi** [comment]
@ansibot thank you
@bcoca No I do not have file named "name.*" except for the name.py/name.pyc file in cryptography package
@alikins i am not sure. I only used fortios_config module in this ansible environment. I have tried my yaml file in ansible 2.5 and it worked. In 2.4.2 it threw this error, which i dont really understand. adding name and register to tasks was supported from the beginning. No clue how to debug either...

**Comment by bcoca** [comment]
The error you get indicates that it is finding a module named 'name', which is hitting a check for 'more than one action in same task' and overriding the property named 'name'.

Try `ansible-doc -l |grep name`

**Comment by AuroraLi** [comment]
thank you @bcoca 
You are absolutely right. I ran the command on two different ansible env
env 1 (which gave me the error):
> $ ansible-doc -l | grep name
bigip_hostname                            Manage the hostname of a BIG-IP.                                                                               
general_name                              UNDOCUMENTED                                                                                                   
hostname                                  Manage hostname                                                                                                
**name                                      UNDOCUMENTED**                                                                                                   
namedtype                                 UNDOCUMENTED                                                                                                   
namedval                                  UNDOCUMENTED                                                                                                   
namespaces                                UNDOCUMENTED                                                                                                   
vmware_guest_find                         Find the folder path(s) for a VM by name or UUID        

env 2 (worked fine):
> ansible-doc -l | grep name
[DEPRECATION WARNING]: DEFAULT_SUDO_EXE option, In favor of become which is a
generic framework . This feature will be removed in version 2.8. Deprecation
warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
bigip_hostname                            Manage the hostname of a BIG-IP
hostname                                  Manage hostname

Same for register:
env1:
> $ ansible-doc -l | grep register
ec2_elb                                   De-registers or registers instances from EC2 ELBs                                                              
ecs_taskdefinition                        register a task definition in ecs                                                                              
elb_instance                              De-registers or registers instances from EC2 ELBs                                                              
iam_mfa_device_facts                      List the MFA (Multi-Factor Authentication) devices registered for a user                                       
**register                                  UNDOCUMENTED**                                                                                                   
rhn_register                              Manage Red Hat Network registration using the `rhnreg_ks' command  




env2:
> $ ansible-doc -l | grep register
[DEPRECATION WARNING]: DEFAULT_SUDO_EXE option, In favor of become which is a generic framework .
This feature will be removed in version 2.8. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
ec2_elb                                   De-registers or registers instances from EC2 ELBs
ecs_taskdefinition                        register a task definition in ecs
elb_instance                              De-registers or registers instances from EC2 ELBs
rhn_register                              Manage Red Hat Network registration using the `rhnre...


I installed ansible in env 1 using pip install from pypi wheel since the server has no internet
I installed ansible in env2 using yum
I guess somehow I hit some corner cases during installation. I will try to reinstall ansible.
